src/lib/normalizer.cc

```diff
@@ -212,13 +220,321 @@ bool Normalizer::VerbalizeSemioticClass(const Token &markup,
+          MutableTransducer output; 
+          Token *token = utt.mutable_linguistic()->mutable_tokens(i);
+          string token_form = ToString(*token);
+            
+          token->set_first_daughter(-1);  // Sets to default unset.
+          token->set_last_daughter(-1);   // Sets to default unset.
+          // Add a single silence for punctuation that forms phrase breaks. This is
+          // set via the grammar, though ultimately we'd like a proper phrasing
+          // module.
+          if (token->type() == Token::PUNCT) {
+            if (token->phrase_break() &&
+                (utt.linguistic().words_size() == 0 ||
+                 utt.linguistic().words(
+                     utt.linguistic().words_size() - 1).id() != "sil")) {
+                
+                /* insert epsilon for punctuation */
+                CompileStringToEpsilon("", &output);
+            }
+          } else if (token->type() == Token::SEMIOTIC_CLASS) {
+            if (!token->skip()) {
+              LoggerDebug("Verbalizing: [%s]\n", token_form.c_str());
+              string words;
+              if (VerbalizeSemioticClass(*token, &output)) {
+              } else {
+                break;
+                LoggerWarn("First-pass verbalization FAILED for [%s]",
+                           token_form.c_str());
+                // Back off to verbatim reading
+                string original_token = token->name();
+                token->Clear();
+                token->set_name(original_token);
+                token->set_verbatim(original_token);
+                if (VerbalizeSemioticClass(*token, &output)) {
+                  LoggerWarn("Reversion to verbatim succeeded for [%s]",
+                             original_token.c_str());
+                } else {
+                  // If we've done our checks right, we should never get here
+                  LoggerError("Verbalization FAILED for [%s]", token_form.c_str());
+              //continue;
+                break;
+                }
+              }
+            }
+          } else if (token->type() == Token::WORD) {
+            if (token->has_wordid()) {
+                /* Writes original string */
+                if (!compiler(token->wordid(), &output)) {
+                    printf("Failed to compile input string \"%s\"", token->wordid().c_str());
+                    //continue;
+                    break;
+                }
+            } else {
+              LoggerError("Token [%s] has type WORD but there is no word id",
+                          token_form.c_str());
+              //continue;
+              break;
+            }
+          } else {
+            LoggerError("No type found for [%s]", token_form.c_str());
+            //continue;
+            break;
+          }
+ 
+         /* Project and Optimize */       
+         MutableTransducer temp;
+         fst::Project(&output, fst::PROJECT_OUTPUT);
+
+         fst::ShortestPath(output, &temp);
```

Comment:
I don't think we want to do this here, if we have cases where the Verbalizer can return multiple paths for one tokenize instance (e.g. for different orderings).